cmd/minikube/cmd/start.go

```diff
@@ -790,6 +790,11 @@ func validateUser(drvName string) {
 		exit.Message(reason.DrvNeedsRoot, `The "{{.driver_name}}" driver requires root privileges. Please run minikube using 'sudo -E minikube start --driver={{.driver_name}}'.`, out.V{"driver_name": drvName})
 	}
 
+	// None driver works with root and without root on Linux
+	if runtime.GOOS == "linux" && drvName == driver.None {
```

Comment:
this PR should check if --interactive flag is set to false (if set to false, means user can not enter anything through stdin and in that case we should exit and advice user that they either have to run with sudo or allow interactive

Reply:
If it helps, there is some existing code you can copy here, that tests (with sudo -n) whether or not sudo requires a prompt, and then exits if interactive mode is disabled:

https://github.com/kubernetes/minikube/blob/5ce9d99d68a57f833b7af035c384479122479968/pkg/minikube/driver/install.go#L96